app/helpers/cart_helper.rb

```diff
@@ -0,0 +1,32 @@
+module CartHelper
+  def get_name_product pro_id
+    product = Product.find_by(id: pro_id)
+    product&.name
+  end
+
+  def get_price_product pro_id
+    product = Product.find_by(id: pro_id)
+    product&.price
+  end
+
+  def get_category_product pro_id
+    product = Product.find_by id: pro_id
+    product&.category_name
+  end
+
+  def count_item
+    @carts.sum{|_pro_id, qty| qty}
+  end
+
+  def total_sum
+    sum = 0
+    products = Product.by_ids(@carts.map(&:first))
+    @carts.each do |pro_id, qty|
+      product = products.find{|pro| pro.id == pro_id.to_i}
```

Comment:
dùng find_by chjuws k dùng find nhé

Reply:
Chị ơi dùng find_by nó sẽ nhả ra exception ạ 
`wrong number of arguments (given 0, expected 1+)`
`      product = products.find_by{|pro| pro.id == pro_id.to_i}
`

Reply:
Find thì raise exception gái.. suy ra chú phải check product of là bao nhiu trc khi lấy nó đi tìm kiếm product

Reply:
em find từ mảng mà chị, mảng làm gì có hàm find_by ạ

Reply:
`product = products.find{|pro| pro.id == pro_id.to_i}` -> find by nà